### ISSUE TITLE
AddRoleTable doesn't actually add everything

### ISSUE THREAD

**Issue by gzuidhof** [dosco/graphjin#139]
What version of Super Graph are you using? `super-graph version`
I'm using Super Graph as a library, v0.14.50.

### Have you tried reproducing the issue with the latest release?
I think this is the latest release

### What is the hardware spec (RAM, OS)?
Not relevant

### Steps to reproduce the issue (config used to run Super Graph).
Create any config with empty Roles and Tables, then:

```golang
 err := conf.AddRoleTable("user", "my_table", core.Query{
 		Limit:   50,
 		Filters: []string{"{ user_id: { eq: $user_id } }"},
 		Columns: []string{"id", "user_id", "created_at", "bla_bla"},
 		Block:   true,
 	})
 
 if err != nil {
 	log.Fatalf("Failed to add role: %v", err)
 }

log.Printf("Tables: %+v\n\n", conf.Tables)
log.Printf("Roles: %+v\n\n", conf.Roles)
```

### Expected behaviour and actual result.
```
Tables: []
Roles: [{Name:user Match: Tables:[] tm:map[]}]
```
The Roles tables field is empty.

I had a look in the `AddRoleTable` function, I expect that it's an issue of by-reference or by-pointer. The newly created Role `r` in the function does get the table added. When I add a print statement at the end of the function:
```
log.Printf("\n\nEnd of function: %+v\n%+v\n", c.Roles[0].Tables, r.Tables)
```

```
End of function: []
[{Name:my_table ReadOnly:false Query:<nil> Insert:<nil> Update:<nil> Delete:<nil>}]
```
Note that it's broken twice in the same way: `Query` is also nil here. 

It took me a while to figure out it wasn't me who was doing something wrong ^^

**Comment by gzuidhof** [comment]
Made a PR for a fix in #140

**Comment by dosco** [comment]
Thanks and sorry for the delay I'm  in a middle of a bit of a major refactor to make things more readable and into smaller packages
